### PR TITLE
New 2.0 API description

### PR DESCRIPTION
**Pull request by nfroidure**
Here starts the new svgicons2svgfont with breaking changes:
- return a duplex stream in order to append glyph to the font progressively (the goal is to be a mirror of https://github.com/nfroidure/svgfont2svgicons )
- change glyphs metadata to a better named property, allow to handle ligatures by handling several codepoints

And new features:
- adding every options to CLI interface
- warn when a font is <1000 of height to avoid all the font distortion issues

Call for advice:
@sapegin @puzrin @sabberworm

### CODE REVIEW

**Comment by nfroidure** [comment]
Currently, only the README is updated so please provide your feedback since it's still just an idea

**Comment by nfroidure** [comment]
Relate https://github.com/nfroidure/svgicons2svgfont/pull/27

File: README.md [hunk 1]
```diff
@@ -16,28 +16,52 @@ You may want to convert fonts to icons, if so use
...
+    console.log('Font successfully created!')
+  })
+  .on('error',function(err) {
+    console.log(err);
   });
 ```
 
-CLI (install the module globally):
+# CLI interface
+All options are available ewcept the `log` one by using this pattern:
```
Comment (sabberworm) [inline README.md]: Typo in `except`.

**Comment by sabberworm** [comment]
I like the API like this. Much better than my suggestion.

How do you envision the integration into gulp-svgicons2svgfont to look like?

**Comment by sabberworm** [comment]
Is the idea that glyphs become ligatures when `codepoints.length > 1`? Why not just write the ligature inside the string: `codepoints: '\uE001\uE002'`? The array notation could then be used to define multiple codepoints for a glyph: `codepoints: ['\uE001\uE002', 'fi']`…

File: README.md [hunk 2]
```diff
@@ -16,28 +16,52 @@ You may want to convert fonts to icons, if so use
...
+var svgicons2svgfont = require('svgicons2svgfont');
+var fs = require('fs');
+var fontStream = svgicons2svgfont({
+  fontName: 'hello'
+});
+
+// Writing glyphs
+var glyph = fs.createReadStream('icons/icon1.svg');
+glyph.metadata = {
+  codepoints: ['0xE001'],
```
Comment (sabberworm) [inline README.md]: Should probably be a number, not a string (lose the quotes).

**Comment by nfroidure** [comment]
@sabberworm i agree with the string form but in order to facilitate the 1.0 to 2.0 migration, i renamed the codepoints option to unicode. Since it is the name of the property in the generated SVG font, it make more sense.
https://github.com/nfroidure/svgicons2svgfont/blob/ea6dc71200ba4f66efce156f4b78f0470c51d470/README.md

**Pull request merged**